pkg/services/ngalert/state/state.go

```diff
@@ -112,10 +113,13 @@ func (a *State) resultNoData(alertRule *ngModels.AlertRule, result eval.Result)
 }
 
 func (a *State) NeedsSending(resendDelay time.Duration) bool {
-	if a.State != eval.Alerting {
```

Comment:
I think we have some good unit tests for NeedsSending, could we add a case or two to make sure this works as expected? 😄